docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,228 @@
+require a longer `TimeoutCommit` to allow the application to complete its work
+and be prepared for the next height.
+
+* From https://github.com/tendermint/spec/issues/359 it looks like `TimeoutCommit` really only exists for the purpose of allowing
+precommits to be collected after the +2/3 has initially been seen. Not much more value to it than that. 
+
+* allowing long timeout commit discussed here: https://github.com/tendermint/tendermint/issues/5911#issuecomment-973560381
+
+## Decision
+
+None
+
+## Detailed Design
+
+### New Consensus Parameters
+
+A new `TimeoutParams` `message` will be added to the [params.proto file][consensus-params-proto].
+This message will have the following form:
+
+```proto
+message TimeoutParams {
+ google.protobuf.Duration timeout_propose = 1
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_propose_delta = 2
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_vote = 3
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_vote_delta = 4
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_commit = 5
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ bool skip_timeout_commit = 6;
+}
+```
+
+This new `message` will then be added as a field into the [`ConsensusParams`
+message][consensus-params-proto]. The same default values that are [currently
+set for these parameters][current-timeout-defaults] in the local configuration
+file will be used as the defaults for these new consensus parameters in the
+[consensus parameter defaults][default-consensus-params].
+
+Validation of these new parameters will be identical to their [validation
+currently][time-param-validation].
+Namely, these parameters must be non-negative.
+
+### Removal of Old Parameters
+
+The old `timeout-*` parameters that are configured in the [config.toml][config-toml]
+will be removed completely.
+
+### Optional: Temporary Local Overrides
+
+The new `TimeoutParams` will be released during the `v0.36` release cycle. Accidentally
+configuring these parameters too low would result could result in chains with slow
+networks or low-degree network topologies to not gossip votes within the configured
+timeouts and require many rounds for consensus to occur. To prevent this condition,
+we could optionally include a set of `unsafe-timeout-*` parameters in the `config.toml`.
+
+These parameters would allow node operators to quickly remedy the situation while
+preparing to update the consensus parameters. This would be a temporary solution that
+we would remove within the v0.37 release.
+
+### Add New Consensus Parameters to HashedParams
+
+Tendermint currently only verifies that a subset of the consensus parameters are
+equal across all validators. These parameters are the `BlockMaxBytes` and the `BlockMaxGas`.
```

Comment:
Do we only verify that these parameters are the same, not any of the others 🤯 ?

Reply:
Yup :) only these ones
https://github.com/tendermint/tendermint/blob/15a3110a6ac2fe769cee0c6cd2c5ad40eabb689c/types/params.go#L185